src/clj/puppetlabs/services/jruby/jruby_puppet_internal.clj

```diff
@@ -110,6 +139,21 @@
         (.put puppet-config dir (fs/absolute-path value))))
     puppet-config))
 
+(schema/defn borrow-with-timeout-fn :- jruby-schemas/JRubyPuppetBorrowResult
+  [timeout :- schema/Int
+   pool :- jruby-schemas/pool-queue-type]
+  (.pollFirst pool timeout TimeUnit/MILLISECONDS))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Public
+
+(schema/defn ^:always-validate
+  create-pool-from-config :- jruby-schemas/PoolState
+  "Create a new PoolData based on the config input."
```

Comment:
You didn't do this, but looks like we forgot to update the docstring when we renamed `PoolData`.  I might update it, but no big deal